### PR TITLE
Add aten::_masked_softmax and its variants

### PR DESCRIPTION
**Pull request by xytintel**
- [x] _masked_softmax
- [x] _masked_softmax_backward

### CODE REVIEW

**Comment by xytintel** [comment]
image
image

**Comment by xytintel** [comment]
image
image

File: src/ATen/native/xpu/sycl/SoftMaxKernels.cpp [hunk 1]
```diff
@@ -1712,6 +1714,221 @@ void spatial_softmax_backward(
...
+        decltype(input_calc)>(                                         \
+        input.data_ptr<scalar_t>(),                                    \
+        output.data_ptr<scalar_t>(),                                   \
+        dim_size,                                                      \
+        outer_size,                                                    \
+        mask.data_ptr<bool>(),                                         \
+        input_calc);                                                   \
+  }
+
+  int max_group_size = 512;
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/SoftMaxKernels.cpp:1774]: Can we have a comment here, if hard code 512. BTW, Can we query it from sycl api

File: src/ATen/native/xpu/sycl/SoftMaxKernels.cpp [hunk 2]
```diff
@@ -1712,6 +1714,221 @@ void spatial_softmax_backward(
...
+        gradInput.data_ptr<scalar_t>(),                     \
+        output.data_ptr<scalar_t>(),                        \
+        gradOutput.data_ptr<scalar_t>(),                    \
+        dim_size,                                           \
+        outer_size,                                         \
+        mask.data_ptr<bool>(),                              \
+        input_calc);                                        \
+  }
+
+  int max_group_size = 512;
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/SoftMaxKernels.cpp:1895]: ditto

**Review by majing921201 — APPROVED** [review]

**Pull request merged**